processor/src/test/java/org/mapstruct/ap/test/defaultvalue/DefaultValueTest.java

```diff
@@ -136,6 +136,10 @@ public void shouldHandleUpdateMethodsFromEntityToEntity() {
                             line = 33,
                             messageRegExp = "Constant and default value are both defined in @Mapping,"
                                     + " either define a defaultValue or a constant." ),
+                    @Diagnostic(type = ErroneousMapper.class,
```

Comment:
The JDK compiler swallows all but the first diagnostic for the same type+line+kind. Eclipse hands out all of them. And we report as much as possible, which I think is good. There are other changes in this PR that also relate to this (e.g. the sorting of the diagnostics).